### PR TITLE
fixes #31457 - Adding method to Task jail

### PR DESCRIPTION
**Pull request by domitea**
This PR is consist of adding `action_output` to Task model and also is added to jail for usage in reports. This methods acts as proxy method for `to_json_output` in Continuous output class. These methods are requested by another PR in REX (https://github.com/theforeman/foreman_remote_execution/pull/453).

### CODE REVIEW

File: lib/foreman_tasks_core/continuous_output.rb [hunk 1]
```diff
@@ -46,5 +46,14 @@ def self.format_output(message, type = 'debug', timestamp = Time.now.getlocal)
         'output' => message,
         'timestamp' => timestamp.to_f }
     end
+
+    def to_json_output
+      sort!
+      outputs_json = { 'outputs': [] }
+      raw_outputs.map do |output|
```
Comment (adamruzicka) [inline lib/foreman_tasks_core/continuous_output.rb:53]: `Enumerable.map` returns an array, using it for side effects isn't exactly best practice.

I'd go with either
```ruby
outputs = raw_outputs.map { |output| { type: output[:output_type], value: output[:message] } }
{ 'outputs': outputs }
```

or just swap `map` for `each` if you want to go the mutating way

Comment (domitea) [reply lib/foreman_tasks_core/continuous_output.rb]: Thanks for the tip! I've updated that.

**Comment by domitea** [comment]
I've made some improvements there. Also I've created a separate PR for template (https://github.com/theforeman/foreman/pull/8178) that is pointing to core.

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
The referenced RM issue is from another project, can we please use: https://projects.theforeman.org/issues/31457 ?
Also, the new method feels somewhat uneccessary now, isn't it?

File: app/models/foreman_tasks/task.rb [hunk 2]
```diff
@@ -242,6 +242,10 @@ def to_label
       parts.join(' ').strip
     end
 
+    def action_output
+      main_action.continuous_output.raw_outputs.sort! if main_action.is_a?(Actions::Helpers::WithContinuousOutput)
```
Comment (ezr-ondrej) [inline app/models/foreman_tasks/task.rb:246]:
```suggestion
      return unless main_action.is_a?(Actions::Helpers::WithContinuousOutput)
      main_action.continuous_output.sort!
      main_action.continuous_output.raw_outputs
```

**Review by ezr-ondrej — APPROVED** [review]
Thanks @domitea ! Looks ok to me :+1:

**Review by adamruzicka — APPROVED** [review]
ACK

**Comment by adamruzicka** [comment]
Thank you @domitea & @ezr-ondrej !

**Pull request merged**